hedera-node/src/main/java/com/hedera/services/store/tokens/BaseTokenStore.java

```diff
@@ -751,9 +792,22 @@ private ResponseCodeEnum checkExistence(AccountID aId, TokenID tId) {
 		return exists(tId) ? OK : INVALID_TOKEN_ID;
 	}
 
+	private ResponseCodeEnum checkExistence(AccountID senderAId, AccountID receiverAId, TokenID tId) {
```

Comment:
Maybe it would be best to split the `checkExistence` in 2
- check for Account Existance
- check for token Existence
that way we can reuse the same methods for both fungible and non fungible.